CONTRIBUTING.md

```diff
@@ -0,0 +1,149 @@
+# Contributing To SwaySwap
+
+Thanks for your interest in contributing to SwaySwap! This document outlines the process for installing dependencies and setting up SwaySwap for development, as well as some conventions on contributing to SwaySwap.
+
+If you run into any difficulties getting started, you can always ask questions on our [Discord](https://discord.gg/xfpK4Pe).
+
+## Setting Up a Development Workspace
+
+A prerequisite for installing and using Sway is the Rust toolchain. Platform-specific instructions can be found [here](https://www.rust-lang.org/tools/install)
+
+Installing `fuel-core` may require installing additional system dependencies. Instructions can be found [here](https://github.com/FuelLabs/fuel-core#building)
+
+A prerequisite for running a Fuel node is Docker. Platform-specific installation instructions can be found [here](https://docs.docker.com/get-docker/)
+
+The Sway toolchain and Fuel Core full node can be installed with:
+
+```sh
+cargo install forc fuel-core
+```
+
+`forc` and `fuel-core` are built and tested against the stable Rust toolchain version `1.5.8` or later. If your install fails the first time, use `rustup update` and try again.
+
+There is no guarantee that either package will work with the `nightly` Rust toolchain, so ensure you are using `stable` with:
+
+```sh
+rustup default stable
+```
+
+This project uses `pnpm` for managing packages and dependencies. `pnpm` can be installed with:
+
+```sh
+npm install -g pnpm
+```
+
+You can install the project dependencies by running the following command in the root directory.
+
+```sh
+pnpm install
+```
+
+A Fuel node can be run locally with:
+
+```sh
+docker compose up
+```
+
+Compile the Sway libraries and contracts located in the `/contracts` directory with:
+
+```sh
+pnpm build-contracts
+```
+
+This command also builds Sway type information for the frontend to use. The types can be found in the `/packages/app/src/types` directory.
+
+Deploy the SwaySwap exhance and token contracts to the Fuel Network locally with:
+
+```sh
+pnpm deploy-contracts
+```
+
+This command also creates a temporary wallet and sends some coins to your wallet. Copy and paste the SwaySwap contract ID and token contract ID into the `.env.example` file.
+
+## Setup Env Variables
+
+After installing the client dependencies in `/packages/app` the contents of `.env.example` are automatically copied to `.env`. In `.env` we have:
+
+- `VITE_FUEL_PROVIDER_URL` - Link for the fuel node
+
+- `VITE_CONTRACT_ID` - Id (address) of the deployed swayswap contract
+
+- `VITE_TOKEN_ID=` - Id (address) of the deployed token contract
+
+- `VITE_RECAPTCHA_SITE_KEY` - The site key is used to invoke recaptcha service on the website
+
+- `VITE_FUEL_FAUCET_URL=` - Link for the fuel faucet
+
+After deploying the SwaySwap and token contracts make sure to copy and paste the SwaySwap and token contract IDs into `.env` before running the development server.
```

Comment:
Maybe change this text to "Before starting the development server make sure the contract id and token id environment variables are set to the corresponding contract addresses." from the app/readme as I believe its slightly more accurate.

Reply:
Will do thx!